vncviewer/vncviewer.cxx

```diff
@@ -503,6 +503,30 @@ static int mktunnel()
 }
 #endif /* !WIN32 */
 
+static int mainloop(const char* vncserver, network::Socket* sock)
```

Comment:
Feels strange to have one function called `mainloop` and another called `run_mainloop`, could you find a different name for this one? Otherwise we should just keep this code in the `main` function.

Reply:
`main` is already quite lengthy so I definitely would vote for putting this in its own method. Good names for the method are welcome, I will also again make up my mind about this

Reply:
My proposal would be to rename `run_mainloop` to `timed_wait` as this is only a wait timer and not a main loop. And than rename my new method `mainloop` into `run_mainloop`. What do you think about this?

Reply:
Yeah, that sounds good to me.